Source/EasyNetQ/Topology/Binding.cs

```diff
@@ -1,24 +0,0 @@
-﻿using System.Collections.Generic;
-
-namespace EasyNetQ.Topology
-{
-    public sealed class Binding : IBinding
-    {
-        public Binding(IBindable bindable, IExchange exchange, string routingKey, IDictionary<string, object> arguments)
-        {
-            Preconditions.CheckNotNull(bindable, "bindable");
-            Preconditions.CheckNotNull(exchange, "exchange");
-            Preconditions.CheckNotNull(routingKey, "routingKey");
-
-            Bindable = bindable;
-            Exchange = exchange;
-            RoutingKey = routingKey;
-            Arguments = arguments;
-        }
-
-        public IBindable Bindable { get; }
```

Comment:
I don't know if it's good to have `ExchangeToQueueBinding` and `ExchangeToExchangeBinding` instead of this `IBindable` or something similar, I agree that an empty interface doesn't feel good, but I think we should push in a single class, because an exchange can have a list of bindings, could the type of binding be an attribute instead?

I'm looking the management ui and it push me to think a single class for the binding:

![image](https://user-images.githubusercontent.com/1633595/103338677-72aa5680-4a7f-11eb-9c7b-29da1f730af7.png)

where you select the type of binding:

![image](https://user-images.githubusercontent.com/1633595/103338687-7b9b2800-4a7f-11eb-8dab-7aa6def53b02.png)

what about something like this:

```csharp
public class Binding<TTo> where TTo : IBindable
{
	public TTo To { get; set; }
	public string RoutingKey { get; set; }
	public IDictionary<string, object> Arguments { get; set; }
}

public interface IBindable
{
	string Name { get; set; }
}

public class Queue : IBindable
{
	public string Name { get; set; }
	public bool IsExclusive { get; }
	// etc...
}

public class Exchange : IBindable
{
	public string Name { get; set; }
	public string Type { get; }
	// etc...
}

void Main()
{
	var bindingToQueue = new Binding<Queue>();
	var bindingToExchange = new Binding<Exchange>();
}
```

in this way you only have one class, the class that is returned from a `Bind()` is a `Binding`, the `IBindable` is used just as a constraint for the generic type parameter (we can also leave that interface empty in this case), and now the `Binding` has full access to the underline `To` with no limitation.

Reply:
Thanks, it was very cool idea. Done